### PR TITLE
Updated pyes and Flask requirements

### PR DESCRIPTION
**Pull request by loleg**
A best-effort to get the project to work again with a current stable version of ElasticSearch #260. Note that I am currently having an issue getting collections to appear in the main search results, and uploaded collections on the user page. User registration, login, collection creation, uploading, has been tested, but my lack of experience with the project has limited my ability to test this patch.

### CODE REVIEW

File: bibserver/search.py [hunk 1]
```diff
@@ -337,8 +337,8 @@ def collection(self):
                 admin = True if metadata != None and auth.collection.update(self.current_user, metadata) else False
                 if metadata and '_display_settings' in metadata:
                     self.search_options.update(metadata['_display_settings'])
-                users = bibserver.dao.Account.query(size=1000000) # pass the userlist for autocomplete admin addition (could be ajax'd)
-                userlist = [i['_source']['_id'] for i in users['hits']['hits']]
+                users = bibserver.dao.Account.query(size=10000) # pass the userlist for autocomplete admin addition (could be ajax'd)
```
Comment (tfmorris) [inline bibserver/search.py]: Is this required by the dependency upgrade or is it an independent change?

File: setup.py [hunk 2]
```diff
@@ -5,10 +5,10 @@
     version = '0.5.1',
```
Comment (tfmorris) [inline setup.py]: Bump the version

File: setup.py [hunk 3]
```diff
@@ -5,10 +5,10 @@
...
     packages = find_packages(),
     install_requires = [
-        "Flask==0.8",
+        "Flask",
```
Comment (tfmorris) [inline setup.py]: Add new version minimums for dependencies?

File: bibserver/dao.py [hunk 4]
```diff
@@ -159,15 +158,15 @@ def bulk_upsert(cls, dataset, state=None):
...
             if '_created' not in data:
                 data['_created'] = datetime.now().strftime("%Y%m%d%H%M%S")
             data['_last_modified'] = datetime.now().strftime("%Y%m%d%H%M%S")
-            
+
             index_result = conn.index(data, db, cls.__type__, urllib.quote_plus(id_), bulk=True)
         # refresh required after bulk index
-        conn.refresh()
-    
+        # conn.refresh()
```
Comment (tfmorris) [inline bibserver/dao.py]: Was this removed on purpose?  If so, delete it and the misleading comment above.

If it's an independent change, move it to a different commit/PR.

File: bibserver/search.py [hunk 5]
```diff
@@ -87,8 +87,8 @@ def implicit_facet(self):
     def collections(self):
         if len(self.parts) == 1:
             if util.request_wants_json():
-                res = bibserver.dao.Collection.query(size=1000000)
-                colls = [i['_source'] for i in res['hits']['hits']]
+                res = bibserver.dao.Collection.query(size=10000)
```
Comment (tfmorris) [inline bibserver/search.py]: Same comment as other size change

File: bibserver/search.py (hunk 1 above)
Comment (loleg) [inline bibserver/search.py]: ES throws an error (might have been a caught warning), asking specifically for that value.

File: bibserver/dao.py (hunk 4 above)
Comment (loleg) [inline bibserver/dao.py]: This was deprecated, no such method.

File: bibserver/search.py (hunk 5 above)
Comment (loleg) [inline bibserver/search.py]: Same response :)

File: bibserver/dao.py (hunk 4 above)
Comment (tfmorris) [inline bibserver/dao.py]: This looks like it was moved to .indices.refresh().

https://pyes.readthedocs.org/en/latest/changelog.html#deprecated

**Comment by rgrp** [comment]
Just want to say a thanks to @loleg for work here and @tfmorris for the great PR review. @tfmorris I believe you are in a position to merge once all is good here.

/cc @danfowler so he knows about this labs project activity: http://okfnlabs.org/projects/bibserver/

**Comment by todrobbins** [comment]
@tfmorris care to review where @loleg ended up?

**Comment by rufuspollock** [comment]
@todrobbins i'd suggest getting this merged. It looked pretty complete.

**Comment by todrobbins** [comment]
@rufuspollock sounds good, but I don't have write access to this repo. 😁

**Comment by todrobbins** [comment]
Thanks all!
